plugins/inputs/kafka_consumer/kafka_consumer.go

```diff
@@ -201,7 +332,7 @@ func (k *KafkaConsumer) Start(acc telegraf.Accumulator) error {
 			handler := NewConsumerGroupHandler(acc, k.MaxUndeliveredMessages, k.parser, k.Log)
 			handler.MaxMessageLen = k.MaxMessageLen
 			handler.TopicTag = k.TopicTag
-			err := k.consumer.Consume(ctx, k.Topics, handler)
+			err := k.consumer.Consume(ctx, k.allWantedTopics, handler)
```

Comment:
I think you need to acquire the lock here to guarantee consistent data...

Reply:
Oooh, this gets a little tricky.  Consume() is a long-running method.  I think you're right that we don't want to be updating the topic list underneath it.

But that means (I think) that a) we should make a copy of k.allWantedTopics (making the copy protected by the lock) and feel that to Consume(), but also that if we update our topic list, we should stop the consumer and restart it with the new list, correct?

Reply:
I would lock, copy, unlock and let consume do it's thing. The list update anyways is not coordinated with `Consume` and thus the list might or might not be old and updated on the way. The alternative is to _always_ update the list before `Consume` and not update it independently...

Reply:
Well, OK, the lock, copy, unlock is what I'm doing now, so....is there anything further that needs doing?